src/containers/AudioRecorderContainer.js

```diff
@@ -335,15 +337,16 @@ class AudioRecorderContainer extends Container {
     // reset the recording duration
     await this.resetRecordingDuration();
 
-    this.record(); // call the record function after setting up to record
-
+    if (wasMicEnabled) {
+      this.record(); // call the record function after setting up to record
+    }
```

Comment:
Actually instead of using this approach, we can refactor the enabled state to be an enum like following: 
```javascript
MicStateEnum = {
INITIAL: 0, // for lack of a better name, go for some other name if that's better
GRANTED: 1,
DENIED: 2
}
```
Also you can create a new file `enums.js` in `src/` and just export this enum out of there so that it can be used across the source

Then instead of returning from `enableMic`, it can just update the state accordingly (state.enabled = MicStateEnum.GRANTED or state.enabled = MicStateEnum.DENIED in the catch block)
```suggestion
    }
```